File: Se226_Lab_6.5_Python/Lab6_5.py
# QUESTION 1 ////////////////////////////////////
list1 = [1, 2, 3, 4, 5]
list2 = [2, 4, 6, 8, 10]

def find_common_elements(list1, list2):
    common_elements = []
    for element in list1:
        if element in list2:
            common_elements.append(element)
    return common_elements

print(find_common_elements(list1, list2))
# QUESTION 2 ////////////////////////////////////
list3 = ["murat", "ada", "kısık", "ahmet"]

def find_palindromes(strings):
    palindrome_strings = []
    for string in strings:
        if string == string[::-1]:
            palindrome_strings.append(string)
    return palindrome_strings

print(find_palindromes(list3))
# QUESTION 3 ////////////////////////////////////
list4 = [2, 3, 4, 5, 6, 7, 8, 12, 13, 17, 21]

def sieve_of_eratosthenes_primes(numbers):
    n = max(numbers)
    prime = [True for i in range(n + 1)]
    p = 2
    while (p * p <= n):
        if prime[p] == True:
            for i in range(p * p, n + 1, p):
                prime[i] = False
        p += 1
    prime_numbers = []
    for i in numbers:
        if i > 1 and prime[i]:
            prime_numbers.append(i)
    return prime_numbers

print(sieve_of_eratosthenes_primes(list4))
#QUESTION 4 ////////////////////////////////////
word_list = ["enlists", "google", "inlets", "banana"]
new_list = []
def anagrams(word, word_list):
    sorted_word = sorted(word)
    anagrams_list = []
    for w in word_list:
        sorted_w = sorted(w.replace(" ", ""))
        if sorted_w == sorted_word:
            anagrams_list.append(w)
    return anagrams_list
print(anagrams("listen",word_list))

File: Se226_Lab6.5_C++/Lab6_5.cpp
#include <iostream>
#include <algorithm>

using namespace std;

///Finding common elements in the list
void find_common_elements(int list1[], int size1, int list2[], int size2) {
    cout << "Common elements are: ";
    for (int i = 0; i < size1; i++) {
        for (int j = 0; j < size2; j++) {
            if (list1[i] == list2[j]) {
                cout << list1[i] << " ";
                break;
            }
        }
    }
    cout << endl;
}

///Finding palindromes
void find_palindromes(string strings[], int size) {
    cout << "Palindromes are: ";
    for (int i = 0; i < size; i++) {
        string str = strings[i];
        string rev_str = str;
        reverse(rev_str.begin(), rev_str.end());
        if (str == rev_str) {
            cout << str << " ";
        }
    }
    cout << endl;
}
///finding primes with algorithm
void sieve_of_eratosthenes_primes(int numbers[], int size) {
    int n = *max_element(numbers, numbers + size);
    bool prime[n + 1];
    fill(prime, prime + n + 1, true);
    int p = 2;
    while (p * p <= n) {
        if (prime[p]) {
            for (int i = p * p; i <= n; i += p) {
                prime[i] = false;
            }
        }
        p++;
    }
    cout << "Prime numbers are: ";
    for (int i = 0; i < size; i++) {
        int num = numbers[i];
        if (num > 1 && prime[num]) {
            cout << num << " ";
        }
    }
    cout << endl;
}

int main() {

    int list1[] = {1, 2, 3, 4, 5,6,12,15};
    int list2[] = {2, 4, 6, 8, 10,12,17};
    int size1 = sizeof(list1) / sizeof(list1[0]);
    int size2 = sizeof(list2) / sizeof(list2[0]);
    find_common_elements(list1, size1, list2, size2);

    string list3[] = {"murat", "ada", "ata", "ahmet"};
    int size3 = sizeof(list3) / sizeof(list3[0]);
    find_palindromes(list3, size3);

}